src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+    """
+    print(f"Calling get_outputs! model_configs: {model_configs}")
+    # model_configs = string_to_list(model_configs)
+    assert input_path.endswith(".json"), "only JSON format is supported"
+    df_rubrics = pd.read_json(input_path)
+    check_df_fields(
+        df_rubrics,
+        # TODO: can we dedup "prompt" and "final_prompt"?
+        required_fields={"prompt", "final_prompt", "category", "additional_information", "raw_completion", "scoring_scales", "criteria", "detailed_analytic_rubric"},
+        # TODO: do we require any of these fields?
+        optional_fields={'annotator', 'reference_answer', 'critique', 'time_per_example', 'clear_goals', 'price_per_example'},
+    )
+    # TODO: parallelize this?
+    for model_config in model_configs:
+        completions = get_model_completions(df_rubrics[["final_prompt", "category"]], model_config, n_to_print=3)
+        df_completions = ae_utils.convert_to_dataframe(completions)
+        df_rubrics = df_rubrics.reset_index(drop=True)
+        df_completions = pd.concat([df_rubrics, df_completions], axis=1)
+        df_completions = df_completions.loc[:, ~df_completions.columns.duplicated()]
+        df_completions.to_json(f"{output_path_prefix}{model_config}.json", orient='records', indent=4)
+
+
+def run_evals(
+    input_path_prefix: str = "output_",
+    output_path_prefix: str = "eval_output_",
+):
+    """
+    For each model, we will:
+    - Load the completions from {input_path_prefix}{model_name}.json file
+    - Save the detailed evaluation results to {output_path_prefix}{model_name}.json file
+    - Save the model card to {output_path_prefix}{model_name}_model_card.json file
+
+    TODO: should we allow customizing evaluator_name (right now it's hardcoded to "gpt4_CoT_v0")?
+    """
+    
+    print(f"Calling run_evals!")
+    input_paths = list_files_with_prefix(input_path_prefix, suffix=".json")
+    assert len(input_paths) > 0, f"No completion files found with prefix {input_path_prefix}!"
+    for input_path in input_paths:
+        print(f"Processing {input_path}...")
+        model_name = input_path.replace(os.path.abspath(input_path_prefix), "").replace(".json", "")
+        eval_result = {}
+        df_completions = pd.read_json(input_path)
+        check_df_fields(
+            df_completions,
+            # TODO: can we dedup "prompt" and "final_prompt"?
+            required_fields={"prompt", "final_prompt", "category", "additional_information", "raw_completion", "scoring_scales", "criteria", "detailed_analytic_rubric"},
+            # TODO: do we require any of these fields?
+            optional_fields={'annotator', 'reference_answer', 'critique', 'time_per_example', 'clear_goals', 'price_per_example', 'output'},
+        )
+        evaluator_name = "gpt4_CoT_v0"
+        df_evaluations = get_evaluations(df_completions, annotators_config=evaluator_name)
+        eval_result["model_name"] = model_name
```

Comment:
print eval results in the terminal? like the leaderboard in alpaca_eval

`print(eval_results.to_string(float_format="%.2f"))`